### PR TITLE
Fixes #5821 (float32 literal comparison issue)

### LINKED ISSUE #5821: float32 literals are translated to double literals in C
this function fails
``` Nim
proc main(): void =
  let y: float32 = 47.11'f32
  doAssert y == 47.11'f32
```

here is the generated C code:
``` C
N_NIMCALL(void, main_UXECz2j6inWSkSqQMQDhsA)(void) {
	NF32 y;
	y = 4.7109999999999999e+01;
	{
		if (!!((y == 4.7109999999999999e+01))) goto LA3_;  // <------ here is the problem		raiseAssert_PpfSiTIeNpMAb9a5o6SglJw_2(((NimStringDesc*) &TM_X9bGGOoNyKehOTF7f2xys3g_2));
	}
	LA3_: ;
}
```

The generated literal is not a 32 bit ``float`` literal like the Nim literal, it does not have the ``f`` suffix that is required for the C float literal, therefore. It is of type double and therefore there will be an implicit conversion to double of y. That is wrong.

I got this bug from this forum entry:
https://forum.nim-lang.org/t/2956

This bug is easy to fix, because simply putting the right suffix for the floating point literal already fixes the problem, as you can see in the example program here:
```C
#include <stdio.h>

int main() {
	float y;
	y = 4.7109999999999999e+01;
  if (y == 4.7109999999999999e+01f) {
    printf("true\n");
  } else {
    printf("false\n");
  }
  if (y == 4.7109999999999999e+01) {
    printf("true\n");
  } else {
    printf("false\n");
  }
}
```
output:
```
true
false
```

### PR DESCRIPTION
**Pull request by dom96**

### CODE REVIEW

File: compiler/ccgexprs.nim [hunk 1]
```diff
@@ -78,8 +78,10 @@ proc genLiteral(p: BProc, n: PNode, ty: PType): Rope =
                         [p.module.tmpBase, rope(id)])
     else:
       result = makeCString(n.strVal)
-  of nkFloatLit..nkFloat64Lit:
+  of nkFloatLit, nkFloat64Lit:
     result = rope(n.floatVal.toStrMaxPrecision)
+  of nkFloat32Lit:
+    result = "((float) $1)" % [rope(n.floatVal.toStrMaxPrecision)]
```
Comment (Araq) [inline compiler/ccgexprs.nim]: ``$1f`` please, produces smaller C code and reads better.

File: compiler/ccgexprs.nim [hunk 2]
```diff
@@ -78,8 +78,10 @@ proc genLiteral(p: BProc, n: PNode, ty: PType): Rope =
                         [p.module.tmpBase, rope(id)])
     else:
       result = makeCString(n.strVal)
-  of nkFloatLit..nkFloat64Lit:
+  of nkFloatLit, nkFloat64Lit:
     result = rope(n.floatVal.toStrMaxPrecision)
+  of nkFloat32Lit:
+    result = rope(n.floatVal.toStrMaxPrecision("f"))
```
Comment (Araq) [inline compiler/ccgexprs.nim]: what's wrong with 

result = rope(n.floatVal.toStrMaxPrecision()) & "f"

Comment (dom96) [inline compiler/ccgexprs.nim]: ``NANf`` is invalid.

Comment (dom96) [inline compiler/ccgexprs.nim]: and ``INFf`` probably too

Comment (Araq) [inline compiler/ccgexprs.nim]: Ok, brilliant you thought of this.

**Pull request merged**